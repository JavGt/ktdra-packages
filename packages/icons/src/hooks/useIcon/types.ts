import { Subs } from '@ktdra-digital/utils';
import { Icon, SVGComponent } from '../../types';

export type UseIcon = (params?: Icon, subsistema?: Subs) => SVGComponent | null;
