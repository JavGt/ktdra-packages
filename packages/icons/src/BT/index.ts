export { default as ActividadSugerida } from './ActividadSugerida';
export { default as Complementate } from './Complementate';
export { default as Conectate } from './Conectate';
export { default as Cuestionate } from './Cuestionate';
export { default as DesarrolloClase } from './DesarrolloClase';
export { default as Experimental } from './Experimental';
export { default as Formativa } from './Formativa';
export { default as GuiaClase } from './GuiaClase';
export { default as Lectora } from './Lectora';
export { default as Lectura } from './Lectura';
export { default as Paginas } from './Paginas';
export { default as ProyectoIntegrador } from './ProyectoIntegrador';
export { default as Recuperacion } from './Recuperacion';
export { default as Tic } from './Tic';
export { default as Tiempo } from './Tiempo';
export { default as Transversal } from './Transversal';
