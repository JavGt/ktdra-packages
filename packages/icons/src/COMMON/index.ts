export { default as Alerta } from './Alerta';
export { default as Barbell } from './Barbell';
export { default as Brain } from './Brain';
export { default as CaraEscalaAlta } from './CaraEscalaAlta';
export { default as CaraEscalaBaja } from './CaraEscalaBaja';
export { default as CaraEscalaMedia } from './CaraEscalaMedia';
export { default as Confetti } from './Confetti';
export { default as DeviceMobile } from './DeviceMobile';
export { default as Devices } from './Devices';
export { default as Download } from './Download';
export { default as Exam } from './Exam';
export { default as Facebook } from './Facebook';
export { default as FlyingSaucer } from './FlyingSaucer';
export { default as MaskHappyThin } from './MaskHappyThin';
export { default as Notebook } from './Notebook';
export { default as Rocke } from './Rocke';
export { default as RocketLaunch } from './RocketLaunch';
export { default as SealCheck } from './SealCheck';
export { default as Student } from './Student';
export { default as Twitter } from './Twitter';
export { default as Youtube } from './Youtube';
