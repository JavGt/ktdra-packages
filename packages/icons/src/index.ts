export * as BT from './BT';
export * as DGB from './DGB';
export * as KTDRA from './KTDRA';
export * as NEM from './NEM';
export * as COMMON from './COMMON';
export * as INGLES from './INGLES';
export * from './hooks';
export * from './types';
export * from './utils';
