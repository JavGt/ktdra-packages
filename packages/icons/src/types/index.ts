export * from './folders';
export * from './icons';
