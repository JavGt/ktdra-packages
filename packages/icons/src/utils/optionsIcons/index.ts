export { default as optionsIcons } from './optionsIcons';
