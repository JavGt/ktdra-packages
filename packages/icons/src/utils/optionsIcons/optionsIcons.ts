// This function filters the ICONS array based on the selected folders.

import { ICONS } from '../../constants';
import { FoldersIcon } from '../../types';

const optionsIcons = (folders: FoldersIcon[]) => {
	if (!folders.length) return ICONS;

	return ICONS.filter((icon) => folders.includes(icon.folder as FoldersIcon));
};

export default optionsIcons;