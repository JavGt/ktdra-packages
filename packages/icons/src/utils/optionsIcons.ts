import { ICONS } from '../constants/icons';

export const optionsIcons = (...args: string[]) => {
	if (!args.length) return ICONS;

	return ICONS.filter((icon) => args.includes(icon.folder));
};
