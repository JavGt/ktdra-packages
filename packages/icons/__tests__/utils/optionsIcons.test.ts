import { FoldersIcon } from '../../src/types';
import { optionsIcons } from '../../src/utils/optionsIcons';

describe('optionsIcons', () => {
  test('should return all icons', () => {
    const folders = [];
    const result = optionsIcons(folders);
    expect(result).toMatchSnapshot();
  });

  test('should return icons with BT', () => {
    const folders = ['BT'] as FoldersIcon[];

    const result = optionsIcons(folders);
    expect(result).toMatchSnapshot();
  });

  test('should return icons with BT and DGB', () => {
    const folders = ['BT', 'DGB'] as FoldersIcon[];

    const result = optionsIcons(folders);
    expect(result).toMatchSnapshot();
  });
});
