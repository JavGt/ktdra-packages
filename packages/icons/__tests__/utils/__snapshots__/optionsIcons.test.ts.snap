// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optionsIcons should return all icons 1`] = `
[
  {
    "folder": "BT",
    "icons": [
      {
        "label": "Actividad sugerida",
        "value": {
          "folder": "BT",
          "name": "ActividadSugerida",
        },
      },
      {
        "label": "Complementate",
        "value": {
          "folder": "BT",
          "name": "Complementate",
        },
      },
      {
        "label": "Conectate",
        "value": {
          "folder": "BT",
          "name": "Conectate",
        },
      },
      {
        "label": "Cuestionate",
        "value": {
          "folder": "BT",
          "name": "Cuestionate",
        },
      },
      {
        "label": "Desarrollo clase",
        "value": {
          "folder": "BT",
          "name": "DesarrolloClase",
        },
      },
      {
        "label": "Experimental",
        "value": {
          "folder": "BT",
          "name": "Experimental",
        },
      },
      {
        "label": "Formativa",
        "value": {
          "folder": "BT",
          "name": "Formativa",
        },
      },
      {
        "label": "Guia clase",
        "value": {
          "folder": "BT",
          "name": "GuiaClase",
        },
      },
      {
        "label": "Lectora",
        "value": {
          "folder": "BT",
          "name": "Lectora",
        },
      },
      {
        "label": "Lectura",
        "value": {
          "folder": "BT",
          "name": "Lectura",
        },
      },
      {
        "label": "Paginas",
        "value": {
          "folder": "BT",
          "name": "Paginas",
        },
      },
      {
        "label": "Proyecto integrador",
        "value": {
          "folder": "BT",
          "name": "ProyectoIntegrador",
        },
      },
      {
        "label": "Recuperacion",
        "value": {
          "folder": "BT",
          "name": "Recuperacion",
        },
      },
      {
        "label": "Tic",
        "value": {
          "folder": "BT",
          "name": "Tic",
        },
      },
      {
        "label": "Tiempo",
        "value": {
          "folder": "BT",
          "name": "Tiempo",
        },
      },
      {
        "label": "Transversal",
        "value": {
          "folder": "BT",
          "name": "Transversal",
        },
      },
    ],
  },
  {
    "folder": "BT/PARTICIPANTES",
    "icons": [
      {
        "label": "Grupo",
        "value": {
          "folder": "BT/PARTICIPANTES",
          "name": "Grupo",
        },
      },
      {
        "label": "Individual",
        "value": {
          "folder": "BT/PARTICIPANTES",
          "name": "Individual",
        },
      },
      {
        "label": "Parejas",
        "value": {
          "folder": "BT/PARTICIPANTES",
          "name": "Parejas",
        },
      },
      {
        "label": "Plenaria",
        "value": {
          "folder": "BT/PARTICIPANTES",
          "name": "Plenaria",
        },
      },
    ],
  },
  {
    "folder": "BT/PLACAS5E",
    "icons": [
      {
        "label": "Elaboramos",
        "value": {
          "folder": "BT/PLACAS5E",
          "name": "Elaboramos",
        },
      },
      {
        "label": "Enganchamos",
        "value": {
          "folder": "BT/PLACAS5E",
          "name": "Enganchamos",
        },
      },
      {
        "label": "Evaluamos",
        "value": {
          "folder": "BT/PLACAS5E",
          "name": "Evaluamos",
        },
      },
      {
        "label": "Explicamos",
        "value": {
          "folder": "BT/PLACAS5E",
          "name": "Explicamos",
        },
      },
      {
        "label": "Exploramos",
        "value": {
          "folder": "BT/PLACAS5E",
          "name": "Exploramos",
        },
      },
    ],
  },
  {
    "folder": "COMMON",
    "icons": [
      {
        "label": "Cara escala alta",
        "value": {
          "folder": "COMMON",
          "name": "CaraEscalaAlta",
        },
      },
      {
        "label": "Cara escala baja",
        "value": {
          "folder": "COMMON",
          "name": "CaraEscalaBaja",
        },
      },
      {
        "label": "Cara escala media",
        "value": {
          "folder": "COMMON",
          "name": "CaraEscalaMedia",
        },
      },
    ],
  },
  {
    "folder": "DGB",
    "icons": [
      {
        "label": "Actividad sugerida",
        "value": {
          "folder": "DGB",
          "name": "ActividadSugerida",
        },
      },
      {
        "label": "Complementate",
        "value": {
          "folder": "DGB",
          "name": "Complementate",
        },
      },
      {
        "label": "Conectate",
        "value": {
          "folder": "DGB",
          "name": "Conectate",
        },
      },
      {
        "label": "Cuestionate",
        "value": {
          "folder": "DGB",
          "name": "Cuestionate",
        },
      },
      {
        "label": "Desarrollo clase",
        "value": {
          "folder": "DGB",
          "name": "DesarrolloClase",
        },
      },
      {
        "label": "Experimental",
        "value": {
          "folder": "DGB",
          "name": "Experimental",
        },
      },
      {
        "label": "Formativa",
        "value": {
          "folder": "DGB",
          "name": "Formativa",
        },
      },
      {
        "label": "Guia clase",
        "value": {
          "folder": "DGB",
          "name": "GuiaClase",
        },
      },
      {
        "label": "Lectora",
        "value": {
          "folder": "DGB",
          "name": "Lectora",
        },
      },
      {
        "label": "Lectura",
        "value": {
          "folder": "DGB",
          "name": "Lectura",
        },
      },
      {
        "label": "Paginas",
        "value": {
          "folder": "DGB",
          "name": "Paginas",
        },
      },
      {
        "label": "Proyecto integrador",
        "value": {
          "folder": "DGB",
          "name": "ProyectoIntegrador",
        },
      },
      {
        "label": "Recuperacion",
        "value": {
          "folder": "DGB",
          "name": "Recuperacion",
        },
      },
      {
        "label": "Tic",
        "value": {
          "folder": "DGB",
          "name": "Tic",
        },
      },
      {
        "label": "Tiempo",
        "value": {
          "folder": "DGB",
          "name": "Tiempo",
        },
      },
      {
        "label": "Transversal",
        "value": {
          "folder": "DGB",
          "name": "Transversal",
        },
      },
    ],
  },
  {
    "folder": "DGB/PARTICIPANTES",
    "icons": [
      {
        "label": "Grupo",
        "value": {
          "folder": "DGB/PARTICIPANTES",
          "name": "Grupo",
        },
      },
      {
        "label": "Individual",
        "value": {
          "folder": "DGB/PARTICIPANTES",
          "name": "Individual",
        },
      },
      {
        "label": "Parejas",
        "value": {
          "folder": "DGB/PARTICIPANTES",
          "name": "Parejas",
        },
      },
      {
        "label": "Plenaria",
        "value": {
          "folder": "DGB/PARTICIPANTES",
          "name": "Plenaria",
        },
      },
    ],
  },
  {
    "folder": "DGB/PLACAS5E",
    "icons": [
      {
        "label": "Elaboramos",
        "value": {
          "folder": "DGB/PLACAS5E",
          "name": "Elaboramos",
        },
      },
      {
        "label": "Enganchamos",
        "value": {
          "folder": "DGB/PLACAS5E",
          "name": "Enganchamos",
        },
      },
      {
        "label": "Evaluamos",
        "value": {
          "folder": "DGB/PLACAS5E",
          "name": "Evaluamos",
        },
      },
      {
        "label": "Explicamos",
        "value": {
          "folder": "DGB/PLACAS5E",
          "name": "Explicamos",
        },
      },
      {
        "label": "Exploramos",
        "value": {
          "folder": "DGB/PLACAS5E",
          "name": "Exploramos",
        },
      },
    ],
  },
  {
    "folder": "INGLES",
    "icons": [
      {
        "label": "Grammar note",
        "value": {
          "folder": "INGLES",
          "name": "GrammarNote",
        },
      },
      {
        "label": "Listening",
        "value": {
          "folder": "INGLES",
          "name": "Listening",
        },
      },
      {
        "label": "Reading",
        "value": {
          "folder": "INGLES",
          "name": "Reading",
        },
      },
      {
        "label": "Speaking",
        "value": {
          "folder": "INGLES",
          "name": "Speaking",
        },
      },
      {
        "label": "Writing",
        "value": {
          "folder": "INGLES",
          "name": "Writing",
        },
      },
    ],
  },
  {
    "folder": "KTDRA",
    "icons": [
      {
        "label": "Logo completo primario",
        "value": {
          "folder": "KTDRA",
          "name": "LogoCompletoPrimario",
        },
      },
      {
        "label": "Logo completo secundario",
        "value": {
          "folder": "KTDRA",
          "name": "LogoCompletoSecundario",
        },
      },
      {
        "label": "Logo texto",
        "value": {
          "folder": "KTDRA",
          "name": "LogoTexto",
        },
      },
      {
        "label": "LogoImage",
        "value": {
          "folder": "KTDRA",
          "name": "LogoImage",
        },
      },
    ],
  },
  {
    "folder": "NEM",
    "icons": [
      {
        "label": "Fomento",
        "value": {
          "folder": "NEM",
          "name": "Fomento",
        },
      },
      {
        "label": "Honestidad",
        "value": {
          "folder": "NEM",
          "name": "Honestidad",
        },
      },
      {
        "label": "Interculturalidad",
        "value": {
          "folder": "NEM",
          "name": "Interculturalidad",
        },
      },
      {
        "label": "Naturaleza",
        "value": {
          "folder": "NEM",
          "name": "Naturaleza",
        },
      },
      {
        "label": "Participacion",
        "value": {
          "folder": "NEM",
          "name": "Participacion",
        },
      },
      {
        "label": "Paz",
        "value": {
          "folder": "NEM",
          "name": "Paz",
        },
      },
      {
        "label": "Respeto",
        "value": {
          "folder": "NEM",
          "name": "Respeto",
        },
      },
      {
        "label": "Responsabilidad",
        "value": {
          "folder": "NEM",
          "name": "Responsabilidad",
        },
      },
    ],
  },
]
`;

exports[`optionsIcons should return icons with BT 1`] = `
[
  {
    "folder": "BT",
    "icons": [
      {
        "label": "Actividad sugerida",
        "value": {
          "folder": "BT",
          "name": "ActividadSugerida",
        },
      },
      {
        "label": "Complementate",
        "value": {
          "folder": "BT",
          "name": "Complementate",
        },
      },
      {
        "label": "Conectate",
        "value": {
          "folder": "BT",
          "name": "Conectate",
        },
      },
      {
        "label": "Cuestionate",
        "value": {
          "folder": "BT",
          "name": "Cuestionate",
        },
      },
      {
        "label": "Desarrollo clase",
        "value": {
          "folder": "BT",
          "name": "DesarrolloClase",
        },
      },
      {
        "label": "Experimental",
        "value": {
          "folder": "BT",
          "name": "Experimental",
        },
      },
      {
        "label": "Formativa",
        "value": {
          "folder": "BT",
          "name": "Formativa",
        },
      },
      {
        "label": "Guia clase",
        "value": {
          "folder": "BT",
          "name": "GuiaClase",
        },
      },
      {
        "label": "Lectora",
        "value": {
          "folder": "BT",
          "name": "Lectora",
        },
      },
      {
        "label": "Lectura",
        "value": {
          "folder": "BT",
          "name": "Lectura",
        },
      },
      {
        "label": "Paginas",
        "value": {
          "folder": "BT",
          "name": "Paginas",
        },
      },
      {
        "label": "Proyecto integrador",
        "value": {
          "folder": "BT",
          "name": "ProyectoIntegrador",
        },
      },
      {
        "label": "Recuperacion",
        "value": {
          "folder": "BT",
          "name": "Recuperacion",
        },
      },
      {
        "label": "Tic",
        "value": {
          "folder": "BT",
          "name": "Tic",
        },
      },
      {
        "label": "Tiempo",
        "value": {
          "folder": "BT",
          "name": "Tiempo",
        },
      },
      {
        "label": "Transversal",
        "value": {
          "folder": "BT",
          "name": "Transversal",
        },
      },
    ],
  },
]
`;

exports[`optionsIcons should return icons with BT and DGB 1`] = `
[
  {
    "folder": "BT",
    "icons": [
      {
        "label": "Actividad sugerida",
        "value": {
          "folder": "BT",
          "name": "ActividadSugerida",
        },
      },
      {
        "label": "Complementate",
        "value": {
          "folder": "BT",
          "name": "Complementate",
        },
      },
      {
        "label": "Conectate",
        "value": {
          "folder": "BT",
          "name": "Conectate",
        },
      },
      {
        "label": "Cuestionate",
        "value": {
          "folder": "BT",
          "name": "Cuestionate",
        },
      },
      {
        "label": "Desarrollo clase",
        "value": {
          "folder": "BT",
          "name": "DesarrolloClase",
        },
      },
      {
        "label": "Experimental",
        "value": {
          "folder": "BT",
          "name": "Experimental",
        },
      },
      {
        "label": "Formativa",
        "value": {
          "folder": "BT",
          "name": "Formativa",
        },
      },
      {
        "label": "Guia clase",
        "value": {
          "folder": "BT",
          "name": "GuiaClase",
        },
      },
      {
        "label": "Lectora",
        "value": {
          "folder": "BT",
          "name": "Lectora",
        },
      },
      {
        "label": "Lectura",
        "value": {
          "folder": "BT",
          "name": "Lectura",
        },
      },
      {
        "label": "Paginas",
        "value": {
          "folder": "BT",
          "name": "Paginas",
        },
      },
      {
        "label": "Proyecto integrador",
        "value": {
          "folder": "BT",
          "name": "ProyectoIntegrador",
        },
      },
      {
        "label": "Recuperacion",
        "value": {
          "folder": "BT",
          "name": "Recuperacion",
        },
      },
      {
        "label": "Tic",
        "value": {
          "folder": "BT",
          "name": "Tic",
        },
      },
      {
        "label": "Tiempo",
        "value": {
          "folder": "BT",
          "name": "Tiempo",
        },
      },
      {
        "label": "Transversal",
        "value": {
          "folder": "BT",
          "name": "Transversal",
        },
      },
    ],
  },
  {
    "folder": "DGB",
    "icons": [
      {
        "label": "Actividad sugerida",
        "value": {
          "folder": "DGB",
          "name": "ActividadSugerida",
        },
      },
      {
        "label": "Complementate",
        "value": {
          "folder": "DGB",
          "name": "Complementate",
        },
      },
      {
        "label": "Conectate",
        "value": {
          "folder": "DGB",
          "name": "Conectate",
        },
      },
      {
        "label": "Cuestionate",
        "value": {
          "folder": "DGB",
          "name": "Cuestionate",
        },
      },
      {
        "label": "Desarrollo clase",
        "value": {
          "folder": "DGB",
          "name": "DesarrolloClase",
        },
      },
      {
        "label": "Experimental",
        "value": {
          "folder": "DGB",
          "name": "Experimental",
        },
      },
      {
        "label": "Formativa",
        "value": {
          "folder": "DGB",
          "name": "Formativa",
        },
      },
      {
        "label": "Guia clase",
        "value": {
          "folder": "DGB",
          "name": "GuiaClase",
        },
      },
      {
        "label": "Lectora",
        "value": {
          "folder": "DGB",
          "name": "Lectora",
        },
      },
      {
        "label": "Lectura",
        "value": {
          "folder": "DGB",
          "name": "Lectura",
        },
      },
      {
        "label": "Paginas",
        "value": {
          "folder": "DGB",
          "name": "Paginas",
        },
      },
      {
        "label": "Proyecto integrador",
        "value": {
          "folder": "DGB",
          "name": "ProyectoIntegrador",
        },
      },
      {
        "label": "Recuperacion",
        "value": {
          "folder": "DGB",
          "name": "Recuperacion",
        },
      },
      {
        "label": "Tic",
        "value": {
          "folder": "DGB",
          "name": "Tic",
        },
      },
      {
        "label": "Tiempo",
        "value": {
          "folder": "DGB",
          "name": "Tiempo",
        },
      },
      {
        "label": "Transversal",
        "value": {
          "folder": "DGB",
          "name": "Transversal",
        },
      },
    ],
  },
]
`;
